employee-rostering-frontend/src/types.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export type Predicate<T> = (value: T) => boolean;
+export type ReadonlyPartial<T> = { readonly [P in keyof T]?: T[P] };
+export type Sorter<T> = (a: T, b: T) => number;
```

Comment:
Good idea! Just please fix imports in tests. Some of them still import the types from the TSX components where they where originally introduced. Running `tsc` from command line can help identify places with TypeScript compilation errors. I have no idea why `tsc` gives different results (fails) than running `npm run build` or `npm test` (so basically react-scripts). There must be some kind of discrepancy between plain `tsconfig.json` and `react-scripts`.

Reply:
Running `tsc` from the command line complains about the `node_modules` folder, and doesn't complain about the wrong imports. I found the locations via a find operation.